File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    let answer =0;
    
    for(let i=1; i<=n; i++){
        if(check(i,0,n)) answer++;
    }
    
    return answer;
}

const check = (start,sum,end)=>{
    if(sum === end) return true;
    if(sum > end) return false;
    return check(start +1,sum+start,end)
}

File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        
        int count = 0;
        int sum = 0;
        
        int start = 0;
        int end = 0;
        
        while(start <= end){
            if(sum == n){
                count++;
                end++;
                sum += end;
            }else if(sum < n){
                end++;
                sum += end;
            }else{
                sum -= start;
                start++;
            }
        }
        return count;
    }
}